apprise/cli.py

```diff
@@ -241,16 +243,58 @@ def main(body, title, config, attach, urls, notification_type, theme, tag,
     # Create our Apprise object
     a = Apprise(asset=asset, debug=debug, location=ContentLocation.LOCAL)
 
-    # Load our configuration if no URLs or specified configuration was
-    # identified on the command line
-    a.add(AppriseConfig(
-        paths=[f for f in DEFAULT_SEARCH_PATHS if isfile(expanduser(f))]
-        if not (config or urls) else config,
-        asset=asset, recursion=recursion_depth))
-
-    # Load our inventory up
-    for url in urls:
-        a.add(url)
+    # The priorities of what is accepted are parsed in order below:
+    #    1. URLs by command line
+    #    2. Configuration by command line
+    #    3. URLs by environment variable: APPRISE_URLS
+    #    4. Configuration by environment variable: APPRISE_CONFIG
+    #    5. Default Configuration File(s) (if found)
+    #
+    if urls:
+        if tag:
+            # Ignore any tags specified
+            logger.warning(
+                '--tag (-g) entries are ignored when using specified URLs')
+            tag = None
+
+        # Load our URLs (if any defined)
+        for url in urls:
+            a.add(url)
+
+        if config:
+            # Provide a warning to the end user if they specified both
+            logger.warning(
+                'You defined both URLs and a --config (-c) entry; '
+                'Only the URLs will be referenced.')
+
+    elif config:
+        # We load our configuration file(s) now only if no URLs were specified
+        # Specified config entries trump all
+        a.add(AppriseConfig(
+            paths=config, asset=asset, recursion=recursion_depth))
+
+    elif os.environ.get('APPRISE_URLS', '').strip():
```

Comment:
I thought of this too, i was just thinking of the Dockerfile/Container environments that pre-define environment variables and leave them empty.  I wanted to allow for this, but not parse it. Thoughts?

Reply:
Although this is the first time I hear about docker setting env vars to empty string if unset, this is a good point. However I think all of `if os.getenv('APPRISE_URLS')`, `if os.getenv('APPRISE_URLS', '')`, `if os.environ.get('APPRISE_URLS')` and `if os.environ.get('APPRISE_URLS', '')` work the same. Because `if x:` evaluates to false for both x = `None` and empty string.

In my personal taste, the call of `.strip()` is unnecessary. It silently ignores the env var if it is set to some form of whitespace. From the perspective of the app this is correct behavior since whitespace-only can't be a valid setting. But for me as a user I would prefer the app I'm using to not silently ignore a non-empty env var. If an env var is set to whitespace-only (wtf?) then something is most likely wrong. But that is just my taste, you decide as you prefer.